auto_submit/lib/service/validation_service.dart

```diff
@@ -88,6 +97,18 @@ class ValidationService {
       Config config, QueryResult result, github.PullRequest messagePullRequest, String ackId, PubSub pubsub) async {
     List<ValidationResult> results = <ValidationResult>[];
 
+    /// Check for the revert label and only process the revert request if the
+    /// revert label is the only label present.
+    List<String> labels = (messagePullRequest.labels as List<github.IssueLabel>)
+        .map<String>((github.IssueLabel labelMap) => labelMap.name)
+        .toList();
+
+    // Process a pull request that is marked with the revert label.
+    if (labels.contains(Config.kRevertLabel) && !labels.contains(Config.kAutosubmitLabel)) {
+      processRevertRequest(config, result, messagePullRequest, ackId, pubsub);
+      return;
+    }
```

Comment:
1) Let's wrap this to a sub function to simplify the logics:
```
if (shouldRevert()) {
  processRevertRequest();
  return;
}
```

This logic sounds a better fit in `processMessage`:
```
if (should revert) {
processRevertRequest;
} else {
ProcessPullRquest;
}
```
2) `messagePullRequest.labels` returns the labels when messaged was pushed to pub/sub. We should obtain the latest labels via an API call.

Reply:
Done.